### PR TITLE
Feature/DR Naming Convention (Alias to be vector name instead.)

### PR DESCRIPTION
**Pull request by JackyKoh**
This PR is about removing the `_dr_.alias.vector_field` naming convention.

This naming convention isnt as necessary for dr vs cluster. It actually helps alleviate multi vector field edge cases of naming as awell.

Original reason why: https://qualitativecloud.slack.com/archives/C027Z6S70RE/p1651826963437089 
This can be a breaking change for tests. @boba-and-beer  should be approved after https://github.com/RelevanceAI/RelevanceAI/pull/743.

### CODE REVIEW

**Comment by JackyKoh** [comment]
@boba-and-beer i screwed up with using black here, will need your help on this one.

**Comment by JackyKoh** [comment]
WIP: update test as well.

File: relevanceai/operations/dr/base.py [hunk 1]
```diff
@@ -32,14 +32,22 @@ def fit(self, *args, **kw):
...
-    def fit_documents(self, vector_field: str, documents: List[Dict]):
-        vectors = self.get_field_across_documents(
-            vector_field, documents, missing_treatment="skip"
+    def fit_documents(self, vector_fields: List[str], documents: List[Dict]):
+        vectors = np.array(
+            self.get_fields_across_documents(
+                vector_fields, documents, missing_treatment="skip"
+            )
         )
+        vectors = vectors.reshape(-1, vectors.shape[1] * vectors.shape[2])
```
Comment (boba-and-beer) [inline relevanceai/operations/dr/base.py:50]: You're flattening the vectors? For DR?

File: relevanceai/operations/dr/base.py [hunk 2]
```diff
@@ -82,20 +90,107 @@ def fit_transform_documents(
...
+
+        documents = list(self.filter_docs_for_fields(vector_fields, documents))
         vectors = np.array(
-            self.get_field_across_documents(
-                vector_field, documents, missing_treatment="skip"
+            self.get_fields_across_documents(
+                vector_fields, documents, missing_treatment="skip"
             )
         )
+        vectors = vectors.reshape(-1, vectors.shape[1] * vectors.shape[2])  # hacky fix
```
Comment (boba-and-beer) [inline relevanceai/operations/dr/base.py:100]: Flattening again? - can you explain what "hacky fix" means here?

File: relevanceai/operations/dr/base.py [hunk 3]
```diff
@@ -82,20 +90,107 @@ def fit_transform_documents(
...
         dr_vectors = self.fit_transform(vectors, dims=dims)
-        dr_vector_field_name = self.get_dr_vector_field_name(vector_field, alias)
-        self.set_field_across_documents(dr_vector_field_name, dr_vectors, documents)
+        del vectors  # free more memory, mainly for memory edgecases
```
Comment (boba-and-beer) [inline relevanceai/operations/dr/base.py:102]: You need to garbage collect after deleting 
```
import gc
gc.collect()
```

**Review by boba-and-beer — CHANGES_REQUESTED** [review]

File: relevanceai/operations/dr/base.py (hunk 1 above)
Comment (JackyKoh) [reply relevanceai/operations/dr/base.py:50]: Yeah, the get_fields_across_documents produces a 3D vector of (len(docs), len(vector_fields), ? or len(vector), if both vector same length))

File: relevanceai/operations/dr/base.py (hunk 2 above)
Comment (JackyKoh) [reply relevanceai/operations/dr/base.py:100]: Flattening only happens once in this call. fit_documents seems to only be fitting not transforming so i didnt use that. hacky because this wont work for vectors of different length atm.

File: relevanceai/operations/dr/base.py (hunk 3 above)
Comment (JackyKoh) [reply relevanceai/operations/dr/base.py:102]: not 100% sure if this is necessary within a function but have added it.

File: relevanceai/operations_new/dr/models/base.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            return f"{self.alias}_vector_"
+        else:
+
+            return f"{self.model_name}_{field}"
+
+    @abstractmethod
+    def fit(self, *args, **kwargs) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
```
Comment (boba-and-beer) [inline relevanceai/operations_new/dr/models/base.py:23]: Delete `fit_predict`

File: relevanceai/operations_new/dr/models/base.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    alias: Union[str, None]
+
+    def vector_name(self, field):
+        if isinstance(self.alias, str):
+            return f"{self.alias}_vector_"
+        else:
+
+            return f"{self.model_name}_{field}"
+
+    @abstractmethod
```
Comment (boba-and-beer) [inline relevanceai/operations_new/dr/models/base.py:19]: Don't think this should be enforced using `abstract`

File: relevanceai/operations_new/dr/models/base.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+from abc import abstractmethod
+
+from typing import Any, Union
+
+from relevanceai.utils import DocUtils
+
+
+class DimReductionModelBase(DocUtils):
```
Comment (boba-and-beer) [inline relevanceai/operations_new/dr/models/base.py:8]:
```
from abc import ABC
class DimReductionBase(DocUtils, ABC)
```

File: relevanceai/operations_new/dr/models/umap.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
+from typing import List, Union
+
+import numpy as np
+
+from relevanceai.operations_new.dr.models.base import DimReductionModelBase
+
+try:
+    from umap import UMAP
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
```
Comment (boba-and-beer) [inline relevanceai/operations_new/dr/models/umap.py:10]: MissingPackageError

File: relevanceai/operations_new/label/base.py [hunk 8]
```diff
@@ -31,11 +31,13 @@
...
 """
-from doc_utils import DocUtils
 from copy import deepcopy
+from typing import Any, Dict, List
 
+from relevanceai.operations_new.base import OperationBase
 
-class LabelBase(DocUtils):
+
+class LabelOperationBase(OperationBase):
```
Comment (boba-and-beer) [inline relevanceai/operations_new/label/base.py:40]: revert to `LabelBase`

File: relevanceai/operations_new/base.py [hunk 9]
```diff
@@ -3,15 +3,13 @@
...
+from typing import Any, Dict, List
 from abc import abstractmethod
 
 from relevanceai.utils import DocUtils
 
 
-class OperationsBase(DocUtils):
-    """Contains API-related functions"""
-
+class OperationBase(DocUtils):
```
Comment (boba-and-beer) [inline relevanceai/operations_new/base.py:12]: make abstractclass

File: relevanceai/operations_new/vectorize/base.py [hunk 10]
```diff
@@ -2,15 +2,19 @@
...
 from abc import abstractmethod
 
-from relevanceai.operations_new.vectorize.models.base import ModelBase
-from relevanceai.operations_new.base import OperationsBase
+from relevanceai.operations_new.vectorize.models.base import VectorizeModelBase
+from relevanceai.operations_new.base import OperationBase
 
 
-class VectorizeBase(OperationsBase):
+class VectorizeOperationBase(OperationBase):
```
Comment (boba-and-beer) [inline relevanceai/operations_new/vectorize/base.py:9]: `VectorizeBase`

File: relevanceai/operations_new/ops.py [hunk 11]
```diff
@@ -126,34 +201,39 @@ def label(
...
+        will be a single vector.
         max_number_of_labels : int, optional
-            The maximum number of labels to return for each document.
+            int = 1,
         similarity_metric : str, optional
-            The metric used to calculate the similarity between the document and the label.
+            str = "cosine",
         filters : Optional[list]
-            A list of filters to apply to the documents.
+            Optional[list] = None,
```
Comment (boba-and-beer) [inline relevanceai/operations_new/ops.py:221]: downgrade

**Review by boba-and-beer — APPROVED** [review]
